server/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -299,34 +334,167 @@ private void updateToLoadAndDrop(List<Notice> notices, Map<String, LookupExtract
   private void takeSnapshot(Map<String, LookupExtractorFactoryContainer> lookupMap)
   {
     if (lookupSnapshotTaker != null) {
-      List<LookupBean> lookups = new ArrayList<>(lookupMap.size());
-      for (Map.Entry<String, LookupExtractorFactoryContainer> e : lookupMap.entrySet()) {
-        lookups.add(new LookupBean(e.getKey(), null, e.getValue()));
+      lookupSnapshotTaker.takeSnapshot(getLookupBeanList(lookupMap));
+    }
+  }
+
+  private void loadAllLookupsAndInitStateRef()
+  {
+    List<LookupBean> lookupBeanList = getLookupsListFromLookupConfig();
+    if (lookupBeanList != null) {
+      startLookups(lookupBeanList);
+    } else {
+      LOG.info("No lookups to be loaded at this point");
+      stateRef.set(new LookupUpdateState(ImmutableMap.of(), ImmutableList.of(), ImmutableList.of()));
+    }
+  }
+
+  /**
+   * Returns a list of lookups from the coordinator if the coordinator is available. If it's not available, returns null.
+   *
+   * @param tier lookup tier name
+   *
+   * @return list of LookupBean objects, or null
+   */
+  @Nullable
+  private List<LookupBean> getLookupListFromCoordinator(String tier)
+  {
+    try {
+      final FullResponseHolder response = fetchLookupsForTier(tier);
+      List<LookupBean> lookupBeanList = new ArrayList<>();
+      if (!response.getStatus().equals(HttpResponseStatus.OK)) {
+        LOG.error(
+            "Error while fetching lookup code from Coordinator with status[%s] and content[%s]",
+            response.getStatus(),
+            response.getContent()
+        );
+        return null;
       }
 
-      lookupSnapshotTaker.takeSnapshot(lookups);
+      // Older version of getSpecificTier returns a list of lookup names.
+      // Lookup loading is performed via snapshot if older version is present.
+      // This check is only for backward compatibility and should be removed in a future release
+      if (response.getContent().startsWith("[")) {
+        return null;
```

Comment:
Please add logging message here, this is a very confusing situation in practice, is just says "coordinator not available"